app/scripts/modules/core/src/presentation/forms/inputs/TextInput.tsx

```diff
@@ -11,6 +12,11 @@ export class TextInput extends React.Component<ITextInputProps> {
   public render() {
     const { value, validation, inputClassName, ...otherProps } = this.props;
     const className = `TextInput form-control ${orEmptyString(inputClassName)} ${validationClassName(validation)}`;
-    return <input className={className} type="text" autoComplete="off" value={orEmptyString(value)} {...otherProps} />;
+    return (
+      <div style={{ display: 'flex', alignItems: 'center' }}>
```

Comment:
This was the best I could come up with to implement this -- clearly I'm not a UI person. 🙁

Reply:
I think the approach (adding a `prefix` prop to `TextInput`) generally makes sense, and the visual treatment is a great start. Few small tweaks I'd suggest:

- To make it less likely we run into weird DOM problems with existing or future usages, it might make more sense to still return the raw `<input>` like we were doing before when there's no `prefix`
- Currently, the `prefix` prop isn't known by the type system, and you're getting around it by using `get` to deref at runtime without actually telling the TypeScript compiler. Instead, please add `prefix` to the ITextInputProps interface as an optional prop and deref it directly
- While inline styles are fine when needed, in this particular case we have some utility classes that are helpful for the styling you're doing. Here's how I'd recommend augmenting your styles:
```tsx
<div className="flex-container-h middle">
  <div className="sp-padding-xs-right body-regular">{get(otherProps, 'prefix') ?? ''}</div>
  <input className={className} type="text" autoComplete="off" value={orEmptyString(value)} {...otherProps} />
</div>
``` 
For the `sp-padding-xs-right` class, note that it uses `4px` instead of `5px`. Hopefully that still looks ok — if it doesn't then just stick with your inline style. for the font size, we usually only use `12px` or `14px` — `body-regular` is `14px` I believe. Either way I recommend picking one of the two instead of using `13px`.